### ISSUE TITLE
Tests: Fix NaN values in gradient tests with ICON physics

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#354]
## Description
Several gradient tests produce NaN values when testing with ICON physics.

## Location
`jcm/model_test.py:327, 351, 418, 420`

## Current Code
```python
# self.assertFalse(jnp.any(jnp.isnan(df_dstate[0].sim_time))) FIXME: this is ending up nan
...
# self.assertFalse(physics_data.isnan().any_true())  FIXME: shortwave_rad has integer value somewhere
```

## Issues
1. `sim_time` gradient ends up NaN in multiple tests
2. `shortwave_rad` has integer value somewhere causing issues

## Impact
Cannot fully validate differentiability of ICON physics.

## Investigation Needed
- Trace source of NaN in sim_time gradient
- Find integer value in shortwave radiation calculation

**Comment by duncanwp** [comment]
Closed by #425
